docker/Dockerfile

```diff
@@ -0,0 +1,44 @@
+ARG PYTHON_VERSION
+FROM python:${PYTHON_VERSION}-slim
+
+RUN apt-get update \
+    && apt-get install -y libmemcached-dev \
+                          build-essential \
+                          libsqlite3-mod-spatialite binutils libproj-dev gdal-bin libgdal20 libgeoip1 \
+                          mysql-client \
+                          unzip libaio1 \
+                          gettext \
+                          wget git \
+    && apt-get clean
+
+RUN groupadd -r test && useradd --no-log-init -r -g test test
+
+RUN mkdir /geolite2/ \
+    && cd /geolite2/ \
+    && wget -q http://geolite.maxmind.com/download/geoip/database/GeoLite2-City.tar.gz \
+        http://geolite.maxmind.com/download/geoip/database/GeoLite2-Country.tar.gz \
+    && (cat *.tar.gz | tar zxvf - --wildcards "*.mmdb" --strip-components=1 -i) \
+    && rm *.tar.gz \
+    && cd /
+
+RUN wget -q https://raw.githubusercontent.com/vishnubob/wait-for-it/master/wait-for-it.sh -O /bin/wait-for-it.sh \
+    && chmod a+x /bin/wait-for-it.sh
+
+ENV PIP_NO_CACHE_DIR=off
+ENV PYTHONDONTWRITEBYTECODE=1
+RUN pip install --upgrade pip
+
+WORKDIR /django
+
+COPY --chown=test:test ./tests/requirements ./tests/requirements
+RUN for f in ./tests/requirements/*.txt; do pip install -r $f; done && \
+    pip install flake8 flake8-isort selenium
+
+COPY --chown=test:test . .
```

Comment:
This is an expensive operation on Docker, is it really necessary ?
(We used that kind of technique and got rid of it recently, for the better).

Reply:
Actually in this case it is not needed at all, as we mount the volumes over it anyway. Nice catch!

It's a pretty expensive operation depending on the number of files, but in most cases it's needed as you want to copy most files into the container. How did you replace it out of interest?

Reply:
Actually, we went from `RUN chown` to `COPY --chown` that doesn't have the cost i was thinking about. We still got rid of it of --chown last month though, building the runtime environment as root because the runtime user should not have write access to modify the scripts. We create directories runtime should write into after the USER command in case there is no volume mounted on target directories. We just ["try not to need chown at all"](https://yourlabs.io/oss/crudlfap/commit/e4df870e9c7d56411d127b06a3d6793664a6935a) but that also came with other cmd reordering so I'm not really sure about the significance of the perf impact (given that we're not talking about RUN chown).

So, my comment was a mistake .. it turns out it's indeed not necessary because compose files have volumes ...

Keep up the great work :+1: